### ISSUE TITLE
Cannot boot on Linux - drivers display crash

### ISSUE THREAD

**Issue by TheYellowArchitect** [godotengine/godot#66272]
### Godot version

4.0 beta1

### System information

Artix Linux (Arch Linux without Systemd) + LARBS (includes dwm)

### Issue description

I want to give some context beforehand. I am a newbie on Linux, and the only reason I haven't jumped fully from Windows 7 (main OS) is that I used to use Unity. With Godot 4.0, I have no reason to use Windows anymore (Unity has Linux versions but my linux build has no spyware/proprietary software)

**I haven't installed any drivers, nor do I know how.** But they may be installed. `mesa` is installed for example.

I tried to find what graphics card I use, so I typed
`lspci -k | grep -EA3 'VGA|3D|Display' which returns:
> 29:00.0 VGA compatible controller: NVIDIA Corporation GP107 [GeForce GTX 1050 Ti] (rev a1)
>              Subsystem: Gigabyte Technology Co., Ltd GV-N105T0C-4GD
>              Kernel driver in use: nouveau
>              Kernel modules: nouveau

So, to the error. I type `./Godot_v4.0-beta1_linux.x86_64 >& error.txt` and the error.txt is:

> Godot Engine v4.0.beta1.official.20d667284 - https://godotengine.org
> ERROR: No surface extension found, is a driver installed?
>    at: _initialize_extensions (drivers/vulkan/vulkan_context.cpp:365)
> ERROR: Could not initialize Vulkan
>    at: DisplayServerX11 (platform/linuxbsd/display_server_x11.cpp:4813)
> Your video card driver does not support any of the supported Vulkan or OpenGL versions.
> Please update your drivers or if you have a very old or integrated GPU, upgrade it.
> If you have updated your graphics drivers recently, try rebooting.
> ERROR: Unable to create DisplayServer, all display drivers failed.
>    at: setup2 (main/main.cpp:1846)
> ERROR: BUG: Unreferenced static string to 0: interface_added
>    at: unref (core/string/string_name.cpp:131)
> ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools11BucketLargeE
>    at: ~PagedAllocator (./core/templates/paged_allocator.h:140)
> ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools12BucketMediumE
>    at: ~PagedAllocator (./core/templates/paged_allocator.h:140)
> ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools11BucketSmallE
>    at: ~PagedAllocator (./core/templates/paged_allocator.h:140)

Note that if I remove the `>& error.txt` the message is the same but its on terminal

Anyway, the error message is clear enough: I must get drivers, so it's possibly not a problem of the engine. Could someone redirect me to a guide on how to install (NVIDIA) drivers on linux? I found [nouveau](https://gitlab.freedesktop.org/nouveau/wiki/-/blob/master/sources/CodeNames.mdwn) where I see the ID of my graphics card is NV137. I also found a [wiki](https://wiki.archlinux.org/title/Nouveau) whose tips I don't know how to use (mesa is already installed lol)

Output to `sudo dmesg | grep 'nouveau'`
> [    3.560613] nouveau 0000:29:00.0: vgaarb: deactivate vga console
> [    3.560671] nouveau 0000:29:00.0: NVIDIA GP107 (137000a1)
> [    3.678290] nouveau 0000:29:00.0: bios: version 86.07.42.00.5b
> [    3.693033] nouveau 0000:29:00.0: pmu: firmware unavailable
> [    3.719407] nouveau 0000:29:00.0: fb: 4096 MiB GDDR5
> [    3.741151] nouveau 0000:29:00.0: DRM: VRAM: 4096 MiB
> [    3.741153] nouveau 0000:29:00.0: DRM: GART: 536870912 MiB
> [    3.741155] nouveau 0000:29:00.0: DRM: BIT table 'A' not found
> [    3.741156] nouveau 0000:29:00.0: DRM: BIT table 'L' not found
> [    3.741157] nouveau 0000:29:00.0: DRM: TMDS table version 2.0
> [    3.741158] nouveau 0000:29:00.0: DRM: DCB version 4.1
> [    3.741159] nouveau 0000:29:00.0: DRM: DCB outp 00: 01000f42 04620030
> [    3.741161] nouveau 0000:29:00.0: DRM: DCB outp 01: 02011f62 04620010
> [    3.741162] nouveau 0000:29:00.0: DRM: DCB outp 02: 02822f76 04600020
> [    3.741163] nouveau 0000:29:00.0: DRM: DCB outp 03: 02022f72 04620020
> [    3.741165] nouveau 0000:29:00.0: DRM: DCB conn 00: 00001031
> [    3.741166] nouveau 0000:29:00.0: DRM: DCB conn 01: 00010161
> [    3.741167] nouveau 0000:29:00.0: DRM: DCB conn 02: 00020246
> [    3.741489] nouveau 0000:29:00.0: DRM: MM: using COPY for buffer copies
> [    3.742632] snd_hda_intel 0000:29:00.1: bound 0000:29:00.0 (ops nv50_audio_component_bind_ops [nouveau])
> [    4.335940] nouveau 0000:29:00.0: DRM: allocated 1366x768 fb: 0x200000, bo 00000000b2bade35
> [    4.336756] fbcon: nouveaudrmfb (fb0) is primary device
> [    4.673633] nouveau 0000:29:00.0: [drm] fb0: nouveaudrmfb frame buffer device
> [    4.710280] [drm] Initialized nouveau 1.3.1 20120801 for 0000:29:00.0 on minor 0

So I assume I already have drivers running? Can someone confirm? So if I have drivers but they dont support Vulkan, I can close this issue. Could be some X11/dwm issue? No idea

### Steps to reproduce

Install [LARBS](https://larbs.xyz/)
Download [Godot](https://downloads.tuxfamily.org/godotengine/4.0/beta1/Godot_v4.0-beta1_linux.x86_64.zip)
`unzip Godot_v4.0-beta1_linux.x86_64.zip`
`./ Godot_v4.0-beta1_linux.x86_64`

**Comment by SeanRamey** [comment]
Well, yeah... this is a graphics driver issue it looks like. Nouveau drivers are just really not very usable with games unfortunately, and Nvidia's drivers are still not fully open source (though they did open source a lot recently). So, you're gonna need proprietary drivers, and that's gonna drop your "no proprietary software" system. But, being as you are basically using a version of arch linux, I recommend installing these packages.

`nvidia-dkms, nvidia-settings, nvidia-utils, and opencl-nvidia`

After installing, reboot, and try again.

**Comment by TheYellowArchitect** [comment]
@SeanRamey I thank you incredibly much. I even posted this problem on an arch linux forum, and all 4 replies didn't help near as much as you did. I almost gave up, and would install Godot on Windows 7, but on my final try, it worked.

The problem mainly was that I couldn't detect what kind of nouveau I had (sudo pacman -Q | grep nouveau -> returned nothing), and I installed the packages you listed, and pretty much replicated this [https://bbs.archlinux.org/viewtopic.php?id=141699 (](https://bbs.archlinux.org/viewtopic.php?pid=1103093#p1103093) and also typed `sudo nvidia-xconfig` twice (first time it failed to read some filepath lol)

tl;dr: Godbless you. I can ditch windows forever now, finally.

(Don't even care I have proprietary software atm, still better than windows. I didn't expect changing drivers to work at all, and expected a black screen from having blacklisted/disabled nouveau and having failed nvidia to render anything. I was ready to grub and save my linux build!)

**Comment by TheYellowArchitect** [comment]
Update: I happened to do a format (again Artix, am so satisfied with this setup), and want to demonstrate the simplified steps (to help anyone who like me lacks opensource NVIDIA drivers support)
`sudo pacman -Syu` (I do it seperately for no network timeout)
`sudo pacman -S nvidia, nvidia-settings, nvidia-utils opencl-nvidia` (no nvidia-dkms)
I didn't uninstall nouveau (sudo pacman -Q | grep nouveau -> returns nothing)
`sudo pacman -S vulkan-icd-loader, vulkan-tools, vulkan-headers, vulkan-validation-layers`
`sudo reboot`
`./godot4.POG`

So no steps required from the linked artix forum, and ofc no need to install `nvidia-xconfig`
